### PR TITLE
Added description for system attribute option

### PR DESCRIPTION
**Pull request by naitsirch**
## Purpose of this pull request

The option reference for EAV attributes did not contain any information about the system option. This PR adds a short description to the reference table.

Additionally a notice was added explaining that a custom attribute is prevented from being added to the `custom_attributes` array if the `system` option is not set to `false`. This was not directly explained previously.

## Affected DevDocs pages

-  https://devdocs.magento.com/guides/v2.3/extension-dev-guide/attributes.html#customer-eav-attribute

## Links to Magento source code

- `Magento\Customer\Model\ResourceModel\Setup\PropertyMapper::map` system option's default is `1`
- `Magento\Customer\Model\Metadata\CustomerMetadata::getCustomAttributesMetadata` prevents system attributes from being added to custom attributes.

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
@naitsirch thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/guides/v2.3/extension-dev-guide/attributes.md [hunk 1]
```diff
@@ -32,7 +32,7 @@ A [module](https://glossary.magento.com/module) has a set of built-in attributes
 *  visibility
 *  weight
 
-In this case, when `getCustomAttributes()` is called, the system returns only custom attributes that are not in this list.
+In this case, when `getCustomAttributes()` is called, the system returns only custom attributes that are not in this list. If you create custom attributes programmatically keep in mind to set the `system` option to false if you want to retrieve the attribute in the `custom_attributes` array.
```
Comment (rogyar) [inline src/guides/v2.3/extension-dev-guide/attributes.md:35]: Hi @naitsirch. Thank you for your collaboration. 
Does it apply to all EAV attributes? I'm currently taking a look at [Product Model](https://github.com/magento/magento2/blob/baebe46dee141a95bf67695ee1bbf4b30af49018/app/code/Magento/Catalog/Model/Product.php#L511), which returns all attributes unless they are blacklisted in the `ProductInterface::ATTRIBUTES` constant without any additional check is the attribute system or not.

Thank you.

Comment (naitsirch) [reply src/guides/v2.3/extension-dev-guide/attributes.md:35]: Looks like you are right. But I thought that line was meant for customer attributes. After reading once again I understand that it is only referred to the catalog module.

The next sentence states 
> The Customer module does not treat its EAV attributes in a special manner. As a result, the getCustomAttributes() method returns all EAV attributes.

So this sentence must be wrong, right? Should I change this sentence and add the hint about the `system` option?

For example like this:
> The Customer module provides a `system` option for its attributes. As a result, the getCustomAttributes() method only returns those EAV attributes that are not defined as system attributes. If you create custom attributes programmatically keep in mind to set the `system` option to false if you want to retrieve the attribute in the `custom_attributes` array.

What do you think?

Comment (rogyar) [reply src/guides/v2.3/extension-dev-guide/attributes.md:35]: That would be awesome. In that way, we will eliminate much of the confusion.

Thank you!

**Comment by naitsirch** [comment]
@rogyar I have updated the PR. Please review

**Comment by dobooth** [comment]
running tests

**Pull request merged**